pydantic_ai_graph/pydantic_ai_graph/state.py

```diff
@@ -25,15 +25,19 @@ def serialize(self) -> bytes | None:
         """Serialize the state object."""
         raise NotImplementedError
```

Comment:
I wonder if we can eliminate this `AbstractState` by moving the serialization and/or copying logic to be kwargs of the graph, and if not provided, use `copy.deepcopy` (or noop if `None` as you've done) for copying, and `pydantic_core.to_json` for serialization. That would let you use a typical basemodel/dataclass/typeddict as state with minimal boilerplate